hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,82 +10,95 @@ This module conforms to the [Comportable guideline for peripheral functionality.
-This means that the FIFO used to collect up the entropy bits will fill, and until the firmware or hardware starts pulling out bits, continuously generated entropy source bits may be dropped from usage.
-The process of filling and draining the FIFOs is the same, independent of what speed the entropy generation rate is at.
-
-Once the `es_entropy_valid` status bit is set, the firmware will read the {{< regref "ES_ENTROPY" >}} register as many times as needed to get the required entropy bits, typically 128 bits or 256 bits at a time.
-For ease of firmware operation, the status bit does not need to be reset.
-A read of the {{< regref "ES_THRESH" >}} register could be done to verify how many entries are in the FIFO at the current time.
-The {{< regref "ES_THRESH" >}} register will determine when the status bit is set, or when the interrupt will assert if enabled.
-For this revision, this same {{< regref "ES_THRESH" >}} register has a range of between 1 and 4 entries in the TLUL FIFO.
-
-For hardware consumption of entropy, if the HWIF FIFO is not empty, and the entropy consuming hardware is ready, then entropy bits will be transferred over the hardware entropy interface.
 
-Health tests, when enabled in the {{< regref "ES_CONF" >}} register, will run continuously.
-Two tests are implemented in hardware, the repetition count test (RCT), and the adaptive proportion test (APT).
-
-There are four health test blocks, one per inbound rng bus bit.
-Each health test block works independently from one another.
-The checks are done on the serial stream input, and an optional feature allows for passing entropy from a single bit stream source.
-Once the streams have passed the health tests, it becomes available for downstream consumption.
-If the health tests ever fail, the ES FIFO is prevented from receiving additional entropy bits up.
-The entropy health tests will continue until a clean stream is detected, at which time the ES FIFO will start collecting bits again.
-The {{< regref "ES_RCT_HEALTH" >}} register and the  {{< regref "ES_APT_HEALTH" >}} register set the cutoff limits for the health tests.
-
-The above process will be repeated for as long as entropy bits are to be collected.
-At any time, the enable bit can be cleared, and the entropy generation will halt immediately.
+After a reset, the ENTROPY_SRC block will start up in bypass mode by default.
+This feature is designed to provide an initial seed's worth of entropy at a faster rate than the normal FIPS/CC compliant entropy process.
+Health testing will still be done on bypass mode entropy, but the window of checking is, by default, 384 bits versus 1024 bits.
+When entropy is delivered to the downstream hardware block, a signal will indicate what type of entropy it is - FIPS compliant or not.
+Bypass mode can be completely disabled in the {{< regref "CONF" >}} register.
+
+Once the initial bypass mode phase has completed, the ENTROPY_SRC block will switch to FIPS compliant mode. 
```

Comment:
is this switch automatic..? or software controlled?

Reply:
There is not an explicit bit, so done by hardware. However, if the entropy control reg is set to route to sw, this could be considered a way to software control the flow. Too convoluted? Open to adding another control bit.

Reply:
I think this should be fine (don't really want to add another control bit and expand the DV space more..)  I think this is somewhat related to my question what should happen if ROM detects some kind of error during bypass phase. 

I feel like ROM should simply move on and leave a note for later handling.. so an auto switch here feels fine to me.  What do you think? 

@moidx 
@gkelly 
@lenary

Reply:
Automatic switching sounds reasonable to me, the less fine-grained control we have to exert in early software the better. Leaving a note I assume is a flag in hardware that we can inspect later?